MiniApp/Classes/MiniAppClient.swift

```diff
@@ -3,21 +3,27 @@ protocol SessionProtocol {
         with request: URLRequest,
         completionHandler: @escaping (Result<ResponseData, Error>) -> Void
     )
+    func startDownloadTask(downloadUrl: URL)
 }
 
-class MiniAppClient {
-    let session: SessionProtocol
+class MiniAppClient: NSObject, URLSessionDownloadDelegate {
+
     let listingApi: ListingApi
     let manifestApi: ManifestApi
+    let downloadApi: DownloadApi
     let environment: Environment
 
-    init(session: SessionProtocol = URLSession(configuration: URLSessionConfiguration.default)) {
-        self.session = session
+    override init() {
         self.environment = Environment()
         self.listingApi = ListingApi(environment: self.environment)
         self.manifestApi = ManifestApi(environment: self.environment)
+        self.downloadApi = DownloadApi(environment: self.environment)
     }
 
+    lazy var session: SessionProtocol = {
+        return URLSession(configuration: .default, delegate: self, delegateQueue: nil)
```

Comment:
I don't think it should _always_ use the default session configuration. The session should probably be injected as a dependency.

Reply:
We had that logic earlier by creating URLSession with default configuration in the constructor. 

`init(session: SessionProtocol = URLSession(configuration: URLSessionConfiguration.default)) `

But now we need to create URLSession object that handles delegates for downloading data task which wouldn't allow to create like below,

`init(session: SessionProtocol = {
        return URLSession(configuration: .default, delegate: self, delegateQueue: nil)
    }()) {`

That's why I was using lazy variable of URLSession. Should I use different APIClient objects for downloading and other?

Reply:
ok, thanks, then I think your way is ok - but maybe revisit this later to see if there's a better approach

Reply:
Yeah sure. May be this could change after MVP once we bring in State machine for downloading.